### PR TITLE
Refactor common scripts

### PR DESCRIPTION
**Pull request by classicalliu**
1. Add `CellCollector` to `secp256k1Blake160`, `secp256k1Blake160Multisig`
2. Refactor `setupInputCell` of `secp256k1Blake160`, `secp256k1Blake160Multisig`, `locktimePool`, `common`
3. Refactor `sudt` to use new `CellCollector`
4. Refactor `injectCapacity` and `transfer` in `common`, can accept customized lock script now.

### CODE REVIEW

File: packages/common-scripts/src/anyone_can_pay.ts [hunk 1]
```diff
@@ -74,23 +76,46 @@ export class CellCollector {
   }
 }
 
-export async function setupInputCell(
+export function setupInputCell(
```
Comment (xxuejie) [inline packages/common-scripts/src/anyone_can_pay.ts]: Even though we might not be using it, let's keep the API interface of setupInputCell an async function for potential needs.

File: packages/common-scripts/src/common.ts [hunk 2]
```diff
@@ -1,142 +1,338 @@
...
+      txSkeleton: TransactionSkeletonType,
+      inputCell: Cell,
+      fromInfo: FromInfo,
+      options: {
+        config?: Config;
+        defaultWitness?: HexString;
+        since?: PackedSince;
+        needCapacity?: HexString;
+      }
+    ): {
```
Comment (xxuejie) [inline packages/common-scripts/src/common.ts]: Let's return a promise here for potentail usage

File: packages/common-scripts/src/common.ts [hunk 3]
```diff
@@ -1,142 +1,338 @@
...
-  tipHeader: Header | undefined,
+  changeAddress?: Address,
+  tipHeader?: Header,
   {
-    config,
-    requireToAddress,
-    useLocktimeCellsFirst,
+    config = undefined,
+    useLocktimeCellsFirst = true,
+    customLockScriptInfos = [],
```
Comment (xxuejie) [inline packages/common-scripts/src/common.ts]: I was thinking maybe we can keep an array in this module, and provide a `registerCustomLockScriptInfo` method which puts custom lockScriptInfo into this global array. Notice that we are building a framework(we are already injecting current config to a global variable anyway), this way developers do not have to pass-in customLockScriptInfos every time they innvoke `transfer`.

Comment (xxuejie) [reply packages/common-scripts/src/common.ts]: In the above design, we also can remove the distinction between predefined lockScriptInfos and customLockScriptsInfos. Predefined ones are just lockScriptInfos that is automatically inserted into the global array.

File: packages/common-scripts/src/anyone_can_pay.ts (hunk 1 above)
Comment (classicalliu) [reply packages/common-scripts/src/anyone_can_pay.ts]: updated.

File: packages/common-scripts/src/common.ts (hunk 2 above)
Comment (classicalliu) [reply packages/common-scripts/src/common.ts]: updated.

File: packages/common-scripts/src/common.ts (hunk 3 above)
Comment (classicalliu) [reply packages/common-scripts/src/common.ts]: updated.

**Review by xxuejie — APPROVED** [review]

**Pull request merged**